src/IROperator.cpp

```diff
@@ -73,6 +73,20 @@ Expr combine_strings(const std::vector<Expr> &args) {
         }
     }
 
+    // Now combine all adjacent string literals, which is
+    // useful to reduce emitted code size when printing
+    size_t i = 0;
+    while (i < strings.size() - 1) {
```

Comment:
Isn't this a for loop?

Reply:
Yes, except that in the 'continue' case we'd need to decrement i, since we want to stay on the same element and the for loop would increment it, and for a size_t that could underflow on the first element, and even if the increment 'fixes' it, it still feels icky and dangerous.

Reply:
Oh right, I missed the continue.

Reply:
...we both missed the 'oops I will crash if passed an empty vector' bug :-(